src/realm/object-store/c_api/app.cpp

```diff
@@ -815,4 +815,16 @@ RLM_API char* realm_user_get_refresh_token(const realm_user_t* user)
     });
 }
 
+RLM_API realm_app_t* realm_user_get_app(const realm_user_t* user)
+{
+    REALM_ASSERT(user);
+    return wrap_err([&]() -> realm_app_t* {
+        if (auto shared_app = (*user)->sync_manager()->app().lock()) {
```

Comment:
Do any of the getters here throw meaningful exceptions? If not, better get rid if `wrap_err`.

Reply:
How do you plan to catch and exception in the user code? Aren't you using C? What about other SDKs that might want to use this?

Reply:
As requested, we are not going to record this error, in case some exception is thrown, the `C API` will discard the exception and return `nullptr` without recording the error. Exceptions can only be thrown if the user has been removed (which is a condition the SDKs should be able to handle correctly).